src/main/java/com/icthh/xm/ms/configuration/topic/ConfigurationUpdateEventProcessor.java

```diff
@@ -0,0 +1,52 @@
+package com.icthh.xm.ms.configuration.topic;
+
+import com.icthh.xm.commons.config.client.repository.message.ConfigurationUpdateMessage;
+import com.icthh.xm.commons.config.domain.Configuration;
+import com.icthh.xm.commons.lep.api.LepManagementService;
+import com.icthh.xm.commons.logging.util.MdcUtils;
+import com.icthh.xm.commons.tenant.TenantContextHolder;
+import com.icthh.xm.commons.tenant.TenantContextUtils;
+import com.icthh.xm.ms.configuration.service.ConcurrentConfigModificationException;
+import com.icthh.xm.ms.configuration.service.ConfigurationService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class ConfigurationUpdateEventProcessor {
+
+    private final TenantContextHolder tenantContextHolder;
+    private final LepManagementService lepManagementService;
+    private final ConfigurationService configurationService;
+
+    public void process(ConfigurationUpdateMessage message, String tenant) {
+        try {
+            init(tenant);
```

Comment:
1 - can we use `tenantContextHolder.getPrivilegedContext().execute(tenantKey, () ->` for run in tenant context

2 - can we use try with resources for `lepManagementService.beginThreadContext` inside tenant context

Reply:
done